src/main/java/org/zalando/riptide/Rest.java

```diff
@@ -69,16 +73,23 @@ public Dispatcher execute(final HttpMethod method, final URI url, final HttpHead
      * response thrown by the {@link OAuth2CompatibilityResponseErrorHandler} and continues with normal dispatching.
      * </p>
      */
-    private <T> ClientHttpResponse execute(final HttpMethod method, final URI url, final Callback<T> callback) {
+    private <T> ClientHttpResponse executeRequest(final HttpMethod method, final URI url, HttpEntity<T> entity) {
         try {
-            return template.execute(url, method, callback, BufferingClientHttpResponse::buffer);
-        } catch (final AlreadyConsumedResponseException e) {
-            return e.getResponse();
+            final ClientHttpRequest request = clientHttpRequestFactory.createRequest(url, method);
+            RequestUtil.writeRequestEntity(entity, request, converters);
+            return request.execute();
+        } catch (IOException ex) {
+            final String message = String.format("I/O error on %s request for \"%s\": %s", method.name(), url, ex.getMessage());
+            throw new ResourceAccessException(message, ex);
         }
     }
 
+    public static Rest create(final ClientHttpRequestFactory clientHttpRequestFactory, final List<HttpMessageConverter<?>> converters) {
+        return new Rest(clientHttpRequestFactory, converters);
+    }
+
     public static Rest create(final RestTemplate template) {
-        return new Rest(template);
+        return create(template.getRequestFactory(), template.getMessageConverters());
```

Comment:
IIRC it offers refreshing the access token transparently. Also the chance that users want to use riptide with a `OAuth2RestTemplate` is higher than the knowledge of the `StupsOAuth2RestTemplate`.